### PR TITLE
show tooltip only above overflowing text

### LINKED ISSUE #5704: More robust tooltip component for overflowing text
and here's a known issue that the tooltip will be cut if the text is too long. i tried the `append-to-body` but seems will cause new issue. i think `NeoTooltip` component should be update so it can adapt more complex scene
also i think there's no precise way to control the tooltip display when the text is overflowing. we may only able to use the text length to control it. like if the text length > 30 then we can display the tooltip.

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5704 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

tooltip above overflowing text only:

image

### <samp>🤖 Generated by Copilot at 50988ef</samp>

Added a composable `useTextOverflow` and a prop `active` for `NeoTooltip` to handle text overflow in collection names. This improves the user experience by showing tooltips only when the text is cut short and avoiding unnecessary re-rendering.

### <samp>🤖 Generated by Copilot at 50988ef</samp>

> _The text is overflowing, the element is cut_
> _We need a composable to handle this rut_
> _`useTextOverflow` is the solution we seek_
> _It gives us `isCut` and `assignRefAndUpdate`_

### CODE REVIEW

**Comment by daiagi** [comment]
@floyd-li  please try it out

**Comment by Jarsen136** [comment]
I wonder if we could manage to see the full text on the tooltip

image

**Comment by daiagi** [comment]
> I wonder if we could manage to see the full text on the tooltip

when #5862 is merged can turn tooltip to multi line

for a better solution i think we will have to wait for
* https://github.com/oruga-ui/oruga/issues/521

**Comment by daiagi** [comment]
It's merged
- [ ]  turn tooltip over popular collection multi-line

**Comment by floyd-li** [comment]
> @floyd-li please try it out

sry for not response in time, just little busy with personal stuff...

**Comment by daiagi** [comment]
>I wonder if we could manage to see the full text on the tooltip

Done, by multi-lining the tooltip

**Comment by exezbcz** [comment]
not sure if related, but the container is too small for this text
image

**Comment by daiagi** [comment]
> not sure if related, but the container is too small for this text image

related.

fix in a couple of minutes

**Comment by exezbcz** [comment]
Fantastic! Perhaps reducing it to less than 4 rows would be worth considering?
image
https://deploy-preview-5919--koda-canary.netlify.app/bsx/gallery/1766361197-2
- something like 2 rows will work i think

**Comment by daiagi** [comment]
@exezbcz 
Done
image

**Comment by exezbcz** [comment]
@daiagi thanks a lot!

File: composables/useTextOverflow.ts
Comment (vikiival) [inline composables/useTextOverflow.ts:1]: What does this feature do sir?

File: components/shared/filters/modules/PopularCollections.vue [hunk 1]
```diff
@@ -32,9 +32,13 @@
...
               class="is-flex is-flex-direction-column is-flex-grow-1 min-width-0">
               <NeoTooltip
+                :active="isCutArray[index].value"
                 :label="collection.meta.name || collection.id"
+                multiline
                 :delay="1000">
-                <div class="is-ellipsis">
+                <div
+                  :ref="(el) => assignRefAndUpdateArray(el, index)"
+                  class="is-ellipsis">
```
Comment (vikiival) [inline components/shared/filters/modules/PopularCollections.vue:41]: is this like magic string cutter ? :eyes:

**Review by Jarsen136 — APPROVED** [review]
If the text overflow logic could be wrapped in the neoTooltip component, that would be better.

**Comment by daiagi** [comment]
> If the text overflow logic could be wrapped in the neoTooltip component, that would be better.

my thought was to make it as a composable so it can be reused in other contexts
however it is quite a niche requirement so maybe you are right and hiding it in neoTooltip is better

ill try and rewrite it

File: composables/useTextOverflow.ts
Comment (vikiival) [reply composables/useTextOverflow.ts:1]: Understood. Thanks

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 25.91 USD/KSM ~ 1.158 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0xf89eff5fef3d234d19fe4032a9eea16f4948ee898a9a914a2477db414202dc1b](https://kusama.subscan.io/extrinsic/0xf89eff5fef3d234d19fe4032a9eea16f4948ee898a9a914a2477db414202dc1b)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**